app/src/main/java/org/mifos/mobile/ui/home/HomeViewModel.kt

```diff
@@ -24,69 +28,72 @@ class HomeViewModel @Inject constructor(private val homeRepositoryImp: HomeRepos
+
+    private val _notificationsCount = MutableStateFlow<Int>(0)
+    val notificationsCount: StateFlow<Int> get() = _notificationsCount
 
     fun loadClientAccountDetails() {
         viewModelScope.launch {
-            _homeUiState.value = HomeUiState.Loading
             homeRepositoryImp.clientAccounts().catch {
                 _homeUiState.value = HomeUiState.Error(R.string.no_internet_connection)
             }.collect { clientAccounts ->
-                _homeUiState.value = HomeUiState.ClientAccountDetails(
-                    getLoanAccountDetails(clientAccounts.loanAccounts),
-                    getSavingAccountDetails(clientAccounts.savingsAccounts)
+                var currentState = (_homeUiState.value as? HomeUiState.Success)?.homeState ?: HomeState()
+                currentState = currentState.copy(
+                    loanAmount = getLoanAccountDetails(clientAccounts.loanAccounts),
+                    savingsAmount = getSavingAccountDetails(clientAccounts.savingsAccounts)
                 )
-
+                _homeUiState.value = HomeUiState.Success(currentState)
             }
         }
     }
 
-
-    val userDetails: Unit
-        get() {
-            viewModelScope.launch {
-                homeRepositoryImp.currentClient().catch {
-                    _homeUiState.value = HomeUiState.Error(R.string.error_fetching_client)
-                }.collect { client ->
-                    preferencesHelper.officeName = client.officeName
-                    _homeUiState.value = HomeUiState.UserDetails(client)
-                }
+    fun getUserDetails() {
+        viewModelScope.launch {
+            homeRepositoryImp.currentClient().catch {
+                _homeUiState.value = HomeUiState.Error(R.string.error_fetching_client)
+            }.collect { client ->
+                preferencesHelper.officeName = client.officeName
+                var currentState = (_homeUiState.value as? HomeUiState.Success)?.homeState ?: HomeState()
+                currentState = currentState.copy(username = client.displayName)
+                _homeUiState.value = HomeUiState.Success(currentState)
             }
         }
+    }
 
-    val userImage: Unit
-        get() {
-            viewModelScope.launch {
-                setUserProfile(preferencesHelper.userProfileImage)
-                homeRepositoryImp.clientImage().catch {
-                    _homeUiState.value = HomeUiState.UserImage(null)
-                }.collect {
-                    val encodedString = it.string()
-                    val pureBase64Encoded =
-                        encodedString.substring(encodedString.indexOf(',') + 1)
-                    preferencesHelper.userProfileImage = pureBase64Encoded
-                    setUserProfile(pureBase64Encoded)
-                }
+    fun getUserImage() {
+        viewModelScope.launch {
+            setUserProfile(preferencesHelper.userProfileImage)
+            homeRepositoryImp.clientImage().catch {}.collect {
```

Comment:
Yes, I have removed the catch block, 
For the bitmap, I have handled in the **MifosUserImage** file itself, that whenever the bitmap is Null, it will call **MifosTextUserImage** which will load the first char of username.

Reply:
No need to dig deep, It's an issue that 404 response is coming. It means you have to handle the exception as well so bring back the catch block and set the bitmap null when comes in catch block and as well as log the response message.

Reply:
Ok, I have logged the response, 
We don't explicitly require to set bitmap to null, as its default value is already null in the UIState, so, the First Char of Username Image will be called.